### DISCUSSION TITLE
Support custom Git hosting provider

### DISCUSSION THREAD

**Discussion by virtadpt** [osmoscraft/osmosmemo#2] [Ideas]
Is ther a way to configure a different Git provider (say, a personal Gitea instance) without modding the source code and recompiling?

**Comment by chuanqisun** [comment]
Currently no easy way to do it as the api endpoints are [hard coded](https://github.com/osmoscraft/osmosmemo/blob/master/src/shared/github/rest-api.ts) against GitHub.

However, I want to support this feature because there is nothing unique to GitHub that other Git hosts cannot provide. I'm converting this to a "Idea discussion" to keep track of it.

**Reply by chuanqisun** [reply]
I think [isomorphic-git](https://isomorphic-git.org/) can be a good candidate library to talk to arbitrary Git hosting provider. If anyone has a suggestion of other tools (or should I go without a library?), please share.
